examples/phi3/phi3_vision.py

```diff
@@ -0,0 +1,362 @@
+
+import argparse
+import glob
+import json
+import os
+import readline
+from pathlib import Path
+
+from olive.common.utils import run_subprocess
+from olive.workflows import run as olive_run
+from olive.workflows.run.config import RunConfig
+
+# flake8: noqa: T201
+# phi3-vision only supports CPU and CUDA targets for now
+TARGETS = ["cpu", "cuda"]
+config_path = Path(__file__).parent / "vision" / "config_templates"
+
+
+def get_args(raw_args):
+    parser = argparse.ArgumentParser(description="phi3 optimization")
+
+    parser.add_argument(
+        "--target",
+        type=str,
+        default="cpu",
+        required=False,
+        choices=TARGETS,
+        help=f"Choose from {TARGETS}",
+    )
+    parser.add_argument(
+        "--precision",
+        type=str,
+        default="int4",
+        choices=["int4"],
+        help=("Precision of quantized model. Currently only int4 is supported. "),
+    )
+    parser.add_argument(
+        "--inference",
+        action="store_true",
+        help="Run inference with optimized model",
+    )
+    parser.add_argument(
+        "--optimized_model_path",
+        type=str,
+        help="Run inference with optimized model",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="cache/phi3-vision-128k-instruct",
+        required=False,
+        help="Path to folder to store ONNX model and additional files (e.g. GenAI config, external data files, etc.)",
+    )
+    parser.add_argument(
+        "--cache_dir",
+        required=False,
+        default="cache",
+        help="Path to cache directory",
+    )
+
+    return parser.parse_args(raw_args)
+
+
+def is_model_ready(input_model_path):
+    if not input_model_path.exists():
+        return False
+    try:
+        from transformers import AutoConfig, AutoModelForCausalLM, AutoProcessor
+
+        config = AutoConfig.from_pretrained(input_model_path, trust_remote_code=True)
+        processor = AutoProcessor.from_pretrained(input_model_path, trust_remote_code=True)
+        model = AutoModelForCausalLM.from_pretrained(input_model_path, trust_remote_code=True)
+        del model, processor, config
+    except Exception as e:
+        print(f"Error loading model: {e}")
+        return False
+    return True
+
+
+def build_text_embedding(args, input_model_path):
+    #########################################
+    # Functions/variables from model builder
+    #########################################
+    import numpy as np
+    import torch
+    from onnx import TensorProto, external_data_helper, helper, numpy_helper, save_model
+    from transformers import AutoConfig, AutoModelForCausalLM
+
+    model = AutoModelForCausalLM.from_pretrained(input_model_path, trust_remote_code=True)
+    config = AutoConfig.from_pretrained(input_model_path, trust_remote_code=True)
+
+    # User inputs
+    io_dtype = TensorProto.FLOAT16 if args.precision == torch.float16 else TensorProto.FLOAT
```

Comment:
args.precision appears to be a string with only "int4" as option. when would this case be true? I think we expect to use fp16 if the target is cuda and fp32 when it is cpu?

Reply:
Updated.